components/analytics/bot-performance-chart.tsx

```
@@ -0,0 +1,373 @@
+      ...platforms
+    }
+  })
+
+  // Data transformation for stacked bar chart showing absolute values
+  const getStackedData = (): StackedDataItem[] => {
+    return chartData.map(day => {
+      const platformData: Record<string, number> = {}
+
+      allPlatforms.forEach(platform => {
+        platformData[platform] = (day[platform] as number) || 0
+      })
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)